win32ss/gdi/eng/engwindow.c

```diff
@@ -163,10 +162,7 @@ EngCreateWnd(
         FIXME("Unsupported flags: 0x%lx\n", fl & ~(WO_RGN_CLIENT_DELTA | WO_RGN_CLIENT | WO_RGN_SURFACE_DELTA | WO_RGN_SURFACE));
     }
 
-    calledFromUser = UserIsEntered();
-    if (!calledFromUser) {
-        UserEnterShared();
-    }
+    UserEnterExclusive();
```

Comment:
Can you move this at the top of the function or remove the whole RETURN thing from this function? If RETURN is used before UserEnterExclusive is used then we have a serious problem.